circe/src/main/scala/org/http4s/circe/middleware/JsonDebugErrorHandler.scala

```diff
@@ -75,13 +75,11 @@ object JsonDebugErrorHandler {
     def encoder[F[_]](
         redactWhen: CIString => Boolean
     ): Encoder[JsonErrorHandlerResponse[F]] =
-      new Encoder[JsonErrorHandlerResponse[F]] {
-        def apply(a: JsonErrorHandlerResponse[F]): Json =
-          Json.obj(
-            "request" -> encodeRequest(a.req, redactWhen),
-            "throwable" -> encodeThrowable(a.caught)
-          )
-      }
+      (a: JsonErrorHandlerResponse[F]) =>
```

Comment:
Could you please say your rationale for making this change, @ayushhooda? Do you find it easier to read, more performant, etc.? 

circe's [Custom encoders/decoders](https://circe.github.io/circe/codecs/custom-codecs.html#custom-encodersdecoders) points to the existing approach, but I admit that I don't know why that is, i.e. as opposed to the approach you've taken in this PR.

Thanks

Reply:
@kevinmeredith
In the majority of cases, lambda expressions could easily replace anonymous classes.
It makes code more **functional**, **precise**, and **readable**.

Reply:
Thanks for providing your rationale!